File: src/features/typing/data/go/jump_game_ii.go
package main

func jump(nums []int) int {
	steps, end, farthest := 0, 0, 0
	for i := 0; i < len(nums)-1; i++ {
		if i+nums[i] > farthest {
			farthest = i + nums[i]
		}
		if i == end {
			steps++
			end = farthest
		}
	}
	return steps
}

File: src/features/typing/data/dart/jump_game_ii.dart
class Solution {
  int jump(List<int> nums) {
    int steps = 0;
    int end = 0;
    int farthest = 0;
    for (int i = 0; i < nums.length - 1; i++) {
      if (i + nums[i] > farthest) {
        farthest = i + nums[i];
      }
      if (i == end) {
        steps++;
        end = farthest;
      }
    }
    return steps;
  }
}
